### PR TITLE
Remove randomkeys dependency implementing it locally

### PR DESCRIPTION
**Pull request by fiunchinho**

### CODE REVIEW

File: service/controller/cloudconfig/cloud_config.go [hunk 1]
```diff
@@ -114,3 +130,84 @@ func newCloudConfig(template string, params k8scloudconfig.Params) (string, erro
...
+	}{
+		{RandomKey: &cluster.APIServerEncryptionKey, Type: EncryptionKey},
+	}
+
+	for _, k := range keys {
+		err := search(ctx, ctrlClient, k.RandomKey, clusterID, namespace, k.Type)
+		if err != nil {
+			return Cluster{}, microerror.Mask(err)
+		}
+	}
```
Comment (tuommaki) [inline service/controller/cloudconfig/cloud_config.go:149]: Is there going to be more elements in `keys` in the future or could we drop this indirection and inline `search()` here?

Comment (fiunchinho) [reply service/controller/cloudconfig/cloud_config.go:149]: I have no idea, tbh. I tried to keep it more or less like it was before, but I don't think the current implementation makes much sense now.

Comment (tuommaki) [reply service/controller/cloudconfig/cloud_config.go:149]: Your call, either way. :slightly_smiling_face:

**Review by tuommaki — APPROVED** [review]

**Review by nprokopic — APPROVED** [review]
LGTM

**Pull request merged**